src/SAWScript/Crucible/LLVM/Builtins.hs

```diff
@@ -728,6 +732,66 @@ verifyPrestate opts cc mspec globals =
 
      return (args, cs, env, globals2)
 
+-- | Checks for contradictions within the given list of assumptions, by asking
+-- the solver about whether their conjunction entails falsehood.
+assumptionsContainContradiction ::
+  (Crucible.HasPtrWidth (Crucible.ArchWidth arch), Crucible.HasLLVMAnn Sym) =>
+  LLVMCrucibleContext arch ->
+  ProofScript SatResult ->
+  [Crucible.LabeledPred Term Crucible.AssumptionReason] ->
+  TopLevel Bool
+assumptionsContainContradiction cc tactic assumptions =
+  do
+     proofGoal <- io $
+      do
+         let sym = cc^.ccBackend
+         st <- readIORef $ W4.sbStateManager sym
+         let sc  = CrucibleSAW.saw_ctx st
+         -- conjunction of all assumptions
+         assume <- scAndList sc (toListOf (folded . Crucible.labeledPred) assumptions)
+         -- implies falsehood
+         goal  <- scImplies sc assume =<< CrucibleSAW.toSC sym (W4.falsePred sym)
```

Comment:
are `goal` and `assume` equivalent?

Reply:
Isn't `goal` the negation of `assume`?

Reply:
of course you are right. Argh, I'm tired...